src/bun.js/process-stdio-polyfill.js

```diff
@@ -0,0 +1,311 @@
+            this.#onFinish = null;
+            cb();
+          }
+        });
+
+        this.#writeStream.on("drain", () => {
+          if (this.#onDrain) {
+            const cb = this.#onDrain;
+            this.#onDrain = null;
+            cb();
+          }
+        });
+
+        eos(this.#writeStream, (err) => {
+          this.#writable = false;
+          if (err) {
+            destroy(this.#writeStream, err);
+          }
+          this.#onFinished(err);
+        });
+
+        this.#readStream.on("ready", () => {
+          if (this.#onReadable) {
+            const cb = this.#onReadable;
+            this.#onReadable = null;
+            cb();
+          } else {
+            this.read();
+          }
+        });
+
+        this.#readStream.on("end", () => {
+          this.push(null);
+        });
+
+        eos(this.#readStream, (err) => {
+          this.#readable = false;
+          if (err) {
+            destroy(this.#readStream, err);
+          }
+          this.#onFinished(err);
+        });
+
+        this.fd = fd;
+      }
+
+      #onFinished(err) {
+        const cb = this.#onClose;
+        this.#onClose = null;
+
+        if (cb) {
+          cb(err);
+        } else if (err) {
+          this.destroy(err);
+        } else if (!this.#readable && !this.#writable) {
+          this.destroy();
+        }
+      }
+
+      _destroy(err, callback) {
+        if (!err && this.#onClose !== null) {
+          err = new AbortError();
+        }
+        this.#onDrain = null;
+        this.#onFinish = null;
+        if (this.#onClose === null) {
+          callback(err);
+        } else {
+          this.#onClose = callback;
+          destroy(this.#writeStream, err);
+          destroy(this.#readStream, err);
+        }
+      }
+
+      _write(chunk, encoding, callback) {
+        if (this.#writeStream.write(chunk, encoding)) {
+          callback();
+        } else {
+          this.#onDrain = callback;
+        }
+      }
+
+      _final(callback) {
+        this.#writeStream.end();
+        this.#onFinish = callback;
+      }
+
+      _read() {
+        while (true) {
+          const buf = this.#readStream.read();
+          if (buf === null || !this.push(buf)) {
+            return;
+          }
+        }
+      }
+    };
+  }
+  return StdioWriteStream;
+}
+
+function getStdinStream({ require }) {
+  if (!StdinStream) {
+    var {
+      Readable,
+      Duplex,
+      eos,
+      destroy,
+    } = require("node:stream", "node:process");
+    if (!createWriteStream) {
+      lazyLoadDeps({ require });
+    }
+
+    StdinStream = class StdinStream extends Duplex {
+      #readStream;
+      #writeStream;
+
+      #readable = true;
+      #writable = true;
+
+      #onFinish;
+      #onClose;
+      #onDrain;
+
+      fd = 0;
+      isTTY = true;
+
+      constructor() {
+        super({ readable: true, writable: true });
+
+        this.#readStream = Readable.fromWeb(Bun.stdin.stream());
```

Comment:
Is this supposed to activate immediately? I thought it had to be explicitly `resume()` or have the handler attached

Reply:
Yeah we can have it activate on adding an event listener for readable or data 👍 I think 
```js
on(ev, listener) {
  super.on(ev, listener);
  if (ev === "data" || ev === "readable")
    this.#readStream = Readable.fromWeb(Bun.stdin.stream());
  // ...rest of setup
}
```
or something like this should work, or what do you think?